sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -755,7 +755,10 @@ case class HashAggregateExec(
     }
 
     // generate hash code for key
-    val hashExpr = Murmur3Hash(groupingExpressions, 42)
+    // SPARK-24076: HashAggregate uses the same hash algorithm on the same expressions
+    // as ShuffleExchange, it may lead to bad hash conflict when shuffle.partitions=8192*n,
+    // pick a different seed to avoid this conflict
+    val hashExpr = Murmur3Hash(groupingExpressions, 48)
```

Comment:
can we just use `UnsafeRow.hashCode` here?

Reply:
@cloud-fan you mean `unsafeRowKeys.hashCode()`, right?
I think it is a good idea, unsafe row has [null bit set] etc., the result should be different, we don't need weird `48` also. Do you want me to create a followup PR?

Reply:
yes please, thanks!

Reply:
@cloud-fan would this perform slower since now we are moving to interpreted version for hashcode generation? If not then why didn't we use `unsafeRowKeys.hashCode()` in the first place?

Reply:
it should be faster, as `unsafeRowKeys.hashCode()` is just one function call. I don't know why we didn't do it in the first place, the code is pretty old.